MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp

```diff
@@ -1530,6 +1532,19 @@ void EnggDiffFittingPresenter::plotFocusedFile(bool plotSinglePeaks) {
     auto focusedPeaksWS =
         ADS.retrieveWS<MatrixWorkspace>(g_focusedFittingWSName);
     auto focusedData = ALCHelper::curveDataFromWs(focusedPeaksWS);
+
+    // Check that the number of curves to plot isn't excessive
+    // lets cap it at 20 to begin with - this number could need
+    // raising but each curve creates about ~5 calls on the stack
+    // so keep the limit low. This will stop users using unfocused
+    // files which have 200+ curves to plot and will "freeze" Mantid
+    constexpr int maxCurves = 20;
```

Comment:
We could double-check this threshold with Saurabh. With the current conventions it could be just 1, as every bank is focused independently and the results written into separate files for every bank. It could also be the number of banks in the instrument (or texture pseudo-banks) if we ever write all the focused results into a single file, but probably never above that number.